src/ol/loadImage.js

```diff
@@ -0,0 +1,50 @@
+
+/**
+ * @typedef {Object} Options
+ * @property {string} [crossOrigin] Cross origin value.
+ * @property {number[]} [size] Size of the image.
+ */
+
+
+/**
+ *
+ * @param {string} src Source.
+ * @param {Options} options Options.
+ * @param {function((HTMLImageElement|ImageBitmap)): any} onSuccess Success .
+ * @param {function(any): any} onError Error callback.
+ */
+export function loadImageUsingDom(src, options, onSuccess, onError) {
+  try {
+    const image = new Image();
```

Comment:
Always using try/catch adds a significant performance penalty.

Reply:
@ahocevar, do you mean it is a performance penalty to have lots of code like the listeners registration in the try catch? I tested with chrome and safari using http://jsben.ch/9E3xn and I see actually the contrary, especially on safari.